### ISSUE TITLE
[FEATURE]

### ISSUE THREAD

**Issue by smiklosovic** [instaclustr/instaclustr-icarus#4]
**Is your feature request related to a problem? Please describe.**

I would like to be able to implement modules and use them without actually merging them into the code base.

**Describe the solution you'd like**

There needs to be a SPI provided / implemented so mere placing of an implementation to class path will result in an operation module being instantiated and registered upon Icarus starting.

**Comment by 2ban** [comment]
Hi, @smiklosovic , any news on this feature? I wanted to list backups via Icarus operations but haven't found one, so I thought implementing my own, but don't really want to change the source code

**Comment by smiklosovic** [comment]
Hi, listing of backups is already there. (1), (2), (3). I am not sure if this module is in Icarus but it should be. I have not tested listing via Icarus. Feel free to test it out and let me know if it works.

(1) https://github.com/instaclustr/esop#listing-of-backups
(2) https://github.com/instaclustr/esop/blob/master/src/main/java/com/instaclustr/esop/impl/list/ListOperation.java
(3) https://github.com/instaclustr/esop/blob/master/src/main/java/com/instaclustr/esop/impl/list/ListModule.java

**Comment by 2ban** [comment]
Yeah, I didn't check for this module in Esop, but it looks like it's not added to Icarus, at least I cannot find it in here https://github.com/instaclustr/icarus/blob/master/src/main/java/com/instaclustr/icarus/Icarus.java

And trying to call this operation to Icarus fails:
Request:
```
curl -X 'POST' \
  'http://http://<icarus>/operations' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '	{
    "type": "list",
    "storageLocation": "file:///tmp/cassandra/dc2/bf652c07-8aff-4cd0-a166-2212f08b0140"
}'
```
Response:
```
can't parse JSON.  Raw result:

Could not resolve type id 'list' as a subtype of `com.instaclustr.operations.OperationRequest`: type ids are not statically known
 at [Source: (org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$UnCloseableInputStream); line: 2, column: 13]
```

**Comment by smiklosovic** [comment]
@2ban could you add it there, test it and create a pull request, please?

**Comment by smiklosovic** [comment]
@2ban I added that module, built JAR is uploaded here:

https://oss.sonatype.org/content/repositories/snapshots/com/instaclustr/icarus/2.0.1-SNAPSHOT/icarus-2.0.1-20211208.183547-1.jar

**Comment by 2ban** [comment]
@smiklosovic great, thanks!
